### DISCUSSION TITLE
GET productsettings request

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2004] [Q&A]
When I select an object in iTwin viewer (created using react template), iTwin seems to generate the following GET request (7- 8 times), all of them returning 403 error.  Each request takes 3-4 secs adding to about 20+ sec delay before the object is selected/highlighted. 

Earlier during startup, there a (seemingly related) PUT request also (below) which also returns 403 error.

Can you please let me know how I can stop iTwin to generate this request.

Thank you,

Ovais

GET https://api.bentley.com/productsettings/v1.0/Application/3098/User/Setting?$filter=namespace+eq+%27imodeljs.presentation%27+and+name+eq+%27FavoriteProperties%27    403 client.js:890 
  push../node_modules/superagent/lib/client.js.Request._end | @ | client.js:890
  push../node_modules/superagent/lib/client.js.Request.end | @ | client.js:744
  (anonymous) | @ | request-base.js:266
   push../node_modules/superagent/lib/request-base.js.RequestBase.then | @ | request-base.js:248

=====The following (related?) request is generated during startup ======================

PUT https://api.bentley.com/productsettings/v1.0/Application/3098/User/Setting/ItwinViewer-SelectionScope/SelectionScope 403

  | push../node_modules/superagent/lib/client.js.Request._end | @ | client.js:890
-- | -- | -- | --
  | push../node_modules/superagent/lib/client.js.Request.end | @ | client.js:744
  | (anonymous) | @ | request-base.js:266
  | push../node_modules/superagent/lib/request-base.js.RequestBase.then | @ | request-base.js:248
  | setTimeout (async) |   |  
  | checkForAdditionalIds | @ | SyncUiEventDispatcher.ts:209
  | setTimeout (async) |   |  
  | dispatchSyncUiEvent | @ | SyncUiEventDispatcher.ts:155
  | dispatchActionToStore | @ | UiFramework.ts:350
  | setActiveSelectionScope | @ | UiFramework.ts:374
  | initializeSelectionScope | @ | SelectionScopeClient.ts:55
  | async function (async) |   |  
  | initializeSelectionScope | @ | SelectionScopeClient.ts:47
  | getModelConnection | @ | IModelLoader.tsx:227
  | async function (async) |   |  
  | getModelConnection | @ | IModelLoader.tsx:175
  | (anonymous) | @ | IModelLoader.tsx:243
  | commitHookEffectListMount | @ | react-dom.development.js:19731
  | commitPassiveHookEffects | @ | react-dom.development.js:19769
  | callCallback | @ | react-dom.development.js:188
  | invokeGuardedCallbackDev | @ | react-dom.development.js:237
  | invokeGuardedCallback | @ | react-dom.development.js:292
  | flushPassiveEffectsImpl | @ | react-dom.development.js:22853
  | unstable_runWithPriority | @ | scheduler.development.js:653
  | runWithPriority$1 | @ | react-dom.development.js:11039
  | flushPassiveEffects | @ | react-dom.development.js:22820
  | (anonymous) | @ | react-dom.development.js:22699
  | workLoop | @ | scheduler.development.js:597
  | flushWork | @ | scheduler.development.js:552
  | performWorkUntilDeadline

**Comment by aruniverse** [comment]
Hi,

This can be safely ignored. The property-grid is making requests to an internal service, product-settings-service, that the viewer doesn't have proper access to. This will be fixed in a future release.

**Reply by ovaisa** [reply]
Thank you Arun for quick response.  The main issue is the 20+ sec delay before the object is selected/highlighted.  We are not using PropertyGrid, is it possible to disable it so that it doesn't generate these errors and associated delay.  I have already disabled using the PropertyGrid defaultUiConfig parameter in Viewer.

**Reply by aruniverse** [reply]
Curious, what version of the itwin-viewer and iModel.js are you using?

This _might_ help preventing the requests from failing and the delay, not sure why you're seeing such a long delay in just selecting elements. 


Create a new file `FavoritePropertiesStorage.ts` with the following:
```ts
import {
  FavoritePropertiesOrderInfo,
  IFavoritePropertiesStorage,
} from "@bentley/presentation-frontend";

export class FavoritePropertiesStorage implements IFavoritePropertiesStorage {
  loadProperties(
    projectId?: string,
    imodelId?: string
  ): Promise<Set<string> | undefined> {
    return Promise.resolve(undefined);
  }
  saveProperties(
    properties: Set<string>,
    projectId?: string,
    imodelId?: string
  ): Promise<void> {
    return Promise.resolve();
  }
  loadPropertiesOrder(
    projectId: string | undefined,
    imodelId: string
  ): Promise<FavoritePropertiesOrderInfo[] | undefined> {
    return Promise.resolve(undefined);
  }
  savePropertiesOrder(
    orderInfos: FavoritePropertiesOrderInfo[],
    projectId: string | undefined,
    imodelId: string
  ): Promise<void> {
    return Promise.resolve();
  }
}
```

And then in your viewer, try to override the [FavoritePropertiesManager](https://www.itwinjs.org/reference/presentation-frontend/core/favoritepropertiesmanager/) :
```ts
...
import {
  FavoritePropertiesManager,
  Presentation,
} from "@bentley/presentation-frontend";
import { FavoritePropertiesStorage } from "./FavoritePropertiesStorage";

  const onIModelAppInit = () => {
    Presentation.setFavoritePropertiesManager(
      new FavoritePropertiesManager({
        storage: new FavoritePropertiesStorage(),
      })
    );
  };

<Viewer
  onIModelAppInit={onIModelAppInit}
  ...
/>
	
```
